train/compute/python/tools/trace_link.py

```diff
@@ -0,0 +1,316 @@
+                kineto_nodes.append(tmp)
+
+    # Max call stack depth when building the tree, the deeper the more accurate but takes longer time
+    depth = 10
+
+    # Build a tree from the kineto trace
+    kineto_graph = transform_to_graph_depth(process_node, depth)
+    logger.info(f"Kineto tree nodes number: {len(kineto_graph.nodes)}")
+
+    # Build a tree from the execution trace
+    et_graph = transform_to_graph_depth(et_nodes[0], depth)
+    logger.info(f"ET tree nodes number: {len(et_graph.nodes)}")
+
+    # Create the GraphMatcher
+    GM = isomorphism.GraphMatcher(kineto_graph, et_graph)
+
+    # Duration of ET nodes
+    et_enhanced = {}
+
+    if GM.is_isomorphic():
+        mapping = GM.mapping
+        logger.info("Graphs are isomorphic")
+        for kineto_id, et_id in mapping.items():
+            et_enhanced[et_id] = kineto_nodes_mapping[kineto_id].end - kineto_nodes_mapping[kineto_id].start
+    else:
+        logger.info("Graphs are not isomorphic")
+
+        # # Compute the edit distance using the graph_edit_distance function with node comparison
+        # paths, cost = nx.graph_edit_distance(kineto_graph, et_graph, node_compare)
+        # logger.info(f"Tree edit distance: {cost}")
+
+        # The problem of finding the exact Graph Edit Distance (GED) is NP-hard so it is often slow
+        # and below is a sub-optimal approach
+
+        edit_distance_generator = nx.optimize_graph_edit_distance(kineto_graph, et_graph, node_compare)
+        cost = next(edit_distance_generator)
+
+        paths_generator = nx.optimize_edit_paths(kineto_graph, et_graph, node_compare)
+        node_edits, edge_edits, cost = next(paths_generator)
+
+        logger.info(f"Sub-optimal tree edit distance: {cost}")
+
+        for kineto_id, et_id in node_edits:
+            if kineto_id is not None and et_id is not None:
+                et_enhanced[et_id] = kineto_nodes_mapping[kineto_id].end - kineto_nodes_mapping[kineto_id].start
+
+    return et_enhanced
+
+
+if __name__ == "__main__":
+    import argparse
+    
+    parser = argparse.ArgumentParser(description="Link kineto trace with execution trace")
+    parser.add_argument("--et-file", type=str, required=True, help="Path to the execution trace")
+    parser.add_argument("--kineto-file", type=str, required=True, help="Path to the kineto trace")
+    parser.add_argument("--annotation", default='DataLoader', type=str, help="Operator name to help slice multiple iterations in trace")
+    parser.add_argument("--exact-match", default=False, action='store_true', help="Whether to match the traces exactly")
+    parser.add_argument("--log-level", default="INFO", help="Log output verbosity.")
+
+    args = parser.parse_args()
+
+    logger.setLevel(args.log_level)
+    
+    et_nodes, kineto_et_events = trace_analysis(args.et_file, args.kineto_file, args.annotation)
+
+    if args.exact_match:
+        et_enhanced = exact_match(kineto_et_events, et_nodes)
```

Comment:
as discussed above this is duration and not the enhance et?
```suggestion
        et_enhanced_duration = exact_match(kineto_et_events, et_nodes)
```